### PR TITLE
Adding server event hooks

### PR DESCRIPTION
**Pull request by ShamsUlAzeem**
This will be useful for examples.

Usage: 
```
KonduitServingMain.builder()
    .onSuccess(() -> {
        // code to run on success
    })
    .onFailure(() -> {
        // code to run on failure
    }).build()
    .runMain(KonduitServingMainArgs.builder()
    .configPath("/path/to/config.json")
    .build().toArgs());
```

### CODE REVIEW

**Comment by ShamsUlAzeem** [comment]
@agibsonccc done!

File: konduit-serving-test/src/test/java/ai/konduit/serving/configprovider/KonduitServingMainTest.java [hunk 1]
```diff
@@ -84,6 +86,72 @@ public void testFile() throws Exception {
         Thread.sleep(10000);
     }
 
+    @Test(timeout = 60000)
+    public void testOnSuccessHook() throws Exception {
+        JsonObject config = getConfig();
+        File jsonConfigPath = new File(System.getProperty("java.io.tmpdir"), "config.json");
```
Comment (agibsonccc) [inline konduit-serving-test/src/test/java/ai/konduit/serving/configprovider/KonduitServingMainTest.java]: Use junit's temporary directory instead: https://howtodoinjava.com/junit/junit-creating-temporary-filefolder-using-temporaryfolder-rule/

File: konduit-serving-test/src/test/java/ai/konduit/serving/configprovider/KonduitServingMainTest.java [hunk 2]
```diff
@@ -84,6 +86,72 @@ public void testFile() throws Exception {
...
+                Assert.fail("onFailure called instead of onSuccess hook");
+                break;
+            }
+        }
+    }
+
+    @Test(timeout = 60000)
+    public void testOnFailureHook() throws Exception {
+        JsonObject config = getConfig();
+        File jsonConfigPath = new File(System.getProperty("java.io.tmpdir"), "config.json");
```
Comment (agibsonccc) [inline konduit-serving-test/src/test/java/ai/konduit/serving/configprovider/KonduitServingMainTest.java]: Same thing re: temp directory above.

File: konduit-serving-test/src/test/java/ai/konduit/serving/configprovider/KonduitServingMainTest.java [hunk 3]
```diff
@@ -84,6 +86,72 @@ public void testFile() throws Exception {
...
+            else {
+                Assert.fail("onSuccess called instead of onFailure hook");
+                break;
+            }
+        }
+    }
+
     public JsonObject getConfig() throws Exception {
         Pair<MultiLayerNetwork, DataNormalization> multiLayerNetwork = TrainUtils.getTrainedNetwork();
         File modelSave = new File(System.getProperty("java.io.tmpdir"), "model.zip");
```
Comment (agibsonccc) [inline konduit-serving-test/src/test/java/ai/konduit/serving/configprovider/KonduitServingMainTest.java]: Same thing.

**Comment by ShamsUlAzeem** [comment]
@agibsonccc done

**Pull request merged**